### PR TITLE
feat(scaffolder): port Bitbucket Server PR action to Bitbucket Cloud

### PR DESCRIPTION
**Pull request by benjidotsh**
This PR ports the Bitbucket Server PR action to Bitbucket Cloud.

The goal was to stay as close to the original code as possible for consistency, so some parts of the code might not be the most elegant, but maybe that's something that should be cleaned up in both actions in another PR.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend-module-bitbucket-cloud/src/actions/bitbucketCloudPullRequest.ts [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/repositories/${workspace}/${repo}/refs/branches`,
+      options,
+    );
+  } catch (e) {
+    throw new Error(`Unable to create branch, ${e}`);
```
Comment (acierto) [inline plugins/scaffolder-backend-module-bitbucket-cloud/src/actions/bitbucketCloudPullRequest.ts:181]: Is `e` here as `string` or `Error`?
If it is Error, it has to be treated differently to show the details of the error properly

File: plugins/scaffolder-backend-module-bitbucket-cloud/src/actions/bitbucketCloudPullRequest.ts [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+
+  const options: RequestInit = {
+    method: 'GET',
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
```
Comment (acierto) [inline plugins/scaffolder-backend-module-bitbucket-cloud/src/actions/bitbucketCloudPullRequest.ts:211]: Can try/catch be removed here?

**Comment by benjidotsh** [comment]
@acierto as mentioned in the description, most of the code (including the lines you commented on) are a straight copy from the Bitbucket Server version. Is this something we want to fix now? I was thinking maybe this should happen in a separate PR for both Bitbucket Cloud and Bitbucket Server.

**Comment by acierto** [comment]
@benjdlambert  wdyt?

**Review by benjdlambert — APPROVED** [review]
I'm fine to merge this as it is for now, but we should deffo create an issue or something to followup with some fixes in the relevant places.

**Pull request merged**